codice/ddf
# DDF-4005 Add profile to run OWASP on DDF distribution and address findings

#### What does this PR do?
Adds `owasp-dist profile`, removes unused jars with vulnerabilities and suppresses invalid ones.

The PR also changes the `Jenkinsfile` to use the `owasp-dist` profile instead of the `owasp` one when building master and release branches.

#### Who is reviewing it? 

@oconnormi 
@mcalcote 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams

#### Ask 2 committers to review/merge the PR and tag them here.
@clockard
@shaundmorris

#### How should this be tested?
Run full normal build, build with `owasp` profile, build with `owasp-dist` profile and make sure they all pass with no vulnerabilities.

Install and run basic ingest and query tests.

#### Any background context you want to provide?
This change has been made to ensure that all the runtime jar files pulled in to build the DDF distribution are scanned and do not contain any vulnerabilities.

#### What are the relevant tickets?
[DDF-4005](https://codice.atlassian.net/browse/DDF-4005)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
